homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -461,7 +473,8 @@ def brightness_received(msg):
 
             device_value = float(payload)
             percent_bright = device_value / self._config[CONF_BRIGHTNESS_SCALE]
-            self._brightness = percent_bright * 255
+            self._attr_brightness = min(round(percent_bright * 255), 255)
```

Comment:
Why `round` if it wasn't there before? 🤔 

Also, why clamp only the max brightness? Is there some other validation that ensures `payload` couldn't have been negative?

Reply:
This is not a refactoring PR, just a step to cleanup properties before applying type hints. Feel free to open a PR.

Reply:
Okay, so why is the non-property-cleanup change here? 😁

Reply:
The properties are removed to reduce the impact of appying type hints later. The _attr_'s are already typed in the super class.

Reply:
I understand. If this PR is only for renaming properties like `_brightness` to `_attr_brightness`, why does this line change in other ways than just the property name?

Reply:
> I understand. If this PR is only for renaming properties like `_brightness` to `_attr_brightness`, why does this line change in other ways than just the property name?

This is not just renaming. The property in the local class is removed.

Reply:
Okay, I think I understand now – the property used to do 
```python
    @property
    def brightness(self):
        """Return the brightness of this light between 0..255."""
        if brightness := self._brightness:
            brightness = min(round(brightness), 255)
        return brightness
```
so that code is brought up here (and there's presumably some other magic that replaces what the property did).

Did I get that right, @jbouwh?

Reply:
You can look at the base Home Assistant Entity classes (LightEntity > ToggleEntity > Entity in this case) to see how the magic happens

Reply:
Thanks for your patience in clearing this up for me :)